src/main/java/de/tum/in/www1/artemis/service/UserService.java

```diff
@@ -418,8 +418,18 @@ public String decryptPassword() {
      * @param login of a user
      * @return decrypted password or empty string
      */
-    public Optional<String> decryptPasswordByLogin(String login) {
-        return userRepository.findOneByLogin(login).map(u -> encryptor().decrypt(u.getPassword()));
+    public Optional<char[]> decryptPasswordByLogin(String login) {
```

Comment:
why do you use char[] and not String?

Reply:
It's best practice to return char arrays instead of Strings when handling passwords in Java since the String then doesn't end up in Java's String pool, or you might have to wait for the Garbage Collector to finally remove the password from memory.

Reply:
I got your point, but this is part of the official jhipster template and not really a part of this PR. We can discuss and change this consistently in a different PR, but I would like to avoid this in this PR, in particular because then you would need to consistently change this to char[] because we use Strings for passwords in several places.

So please change it back!

Reply:
I reverted the changes